src/supervised/other.jl

```diff
@@ -7,15 +7,16 @@ examples and `0` otherwise. It is a generalization of
 """
 struct MisclassLoss <: SupervisedLoss end
 
-agreement(target, output) = target == output
+# return floating point to avoid big integers in aggregations
+value(::MisclassLoss, agreement::Bool) = ifelse(agreement, 0.0, 1.0)
+deriv(::MisclassLoss, agreement::Bool) = 0.0
+deriv2(::MisclassLoss, agreement::Bool) = 0.0
 
-value(::MisclassLoss, agreement::Bool) = agreement ? 0 : 1
-deriv(::MisclassLoss, agreement::Bool) = 0
-deriv2(::MisclassLoss, agreement::Bool) = 0
+const CV = Union{CategoricalValue,Number}
 
-value(loss::MisclassLoss, target::Number, output::Number) = value(loss, agreement(target, output))
-deriv(loss::MisclassLoss, target::Number, output::Number) = deriv(loss, agreement(target, output))
-deriv2(loss::MisclassLoss, target::Number, output::Number) = deriv2(loss, agreement(target, output))
+value(loss::MisclassLoss, target::CV, output::CV) = value(loss, target == output)
```

Comment:
Does it break anything if you remove the `CV` type annotation?

Reply:
It does unfortunately. It introduces an ambiguity as explained in the comments below.